### DISCUSSION TITLE
How to use transactions in LUCID ORM on update and insert

### DISCUSSION THREAD

**Discussion by farshadfahimi** [adonisjs/core#1735] [Help] [answered]
I have a simple code like below and need to store some data with relation but I need to use transaction.
But It seems I should use transaction to store the data not the model.

How I should implement the below code with transaction;

```javascript
await Database.transaction(async (trx) => {
                const bundle = await Bundle.create({
                    title: data.title,
                    user_id: user.id,
                    strategy: data.strategy,
                    started_at: data.started_at,
                    ended_at: data.ended_at,
                    published_at: data.is_published ? new Date : null
                })

                await bundle.bundleProduct().createMany(products)
            })
```

this code throw an error but at last it store the data in database. how I should prevent from this action

**Comment by thetutlage** [comment, marked as answer]
You need to pass the `trx` as 2nd argument to the `createMany` method. https://github.com/adonisjs/lucid/blob/e0b5a316875cf99d4cf5a6f76a103aefb5b945b9/src/Lucid/Model/index.js#L389

**Reply by farshadfahimi** [reply]
I think it's better to put it in the document.

**Reply by thetutlage** [reply]
Sure. You can create a PR for same?
